src/components/__tests__/Note.ts

```diff
@@ -1,110 +1,530 @@
-  ])
+      }),
+      // this will hide the meta attribute, so if the note value can be selected on the screen it must be rendered
+      setCursor(null),
+    ])
 
-  await act(vi.runOnlyPendingTimersAsync)
+    await act(vi.runOnlyPendingTimersAsync)
 
-  const element = screen.getByText('foo')
-  expect(element)
-})
+    // Verify note is rendered
+    const noteElement = screen.queryByLabelText('note')
+    expect(noteElement).toBeInTheDocument()
+
+    // Verify note content
+    const element = screen.getByText('foo')
+    expect(element)
+  })
 
-it('re-render note when =note subthought value changes', async () => {
-  await dispatch([
-    importText({
-      text: `
+  test('re-render note when =note subthought value changes', async () => {
+    await dispatch([
+      importText({
+        text: `
       - a
         - =note
           - foo`,
-    }),
-    setCursor(['a', '=note', 'foo']),
-    (dispatch, getState) =>
-      dispatch(
-        editThought({
-          oldValue: 'foo',
-          newValue: 'bar',
-          path: getState().cursor as SimplePath,
-        }),
-      ),
-    // this will hide the meta attribute, so if the note value can be selected on the screen it must be rendered
-    setCursor(null),
-  ])
-
-  await act(vi.runOnlyPendingTimersAsync)
-
-  const element = screen.getByText('bar')
-  expect(element)
-})
+      }),
+      setCursor(['a', '=note', 'foo']),
+      (dispatch, getState) =>
+        dispatch(
+          editThought({
+            oldValue: 'foo',
+            newValue: 'bar',
+            path: getState().cursor as SimplePath,
+          }),
+        ),
+      // this will hide the meta attribute, so if the note value can be selected on the screen it must be rendered
+      setCursor(null),
+    ])
 
-it('render note when subthought is edited from non-attribute', async () => {
-  await dispatch([
-    importText({
-      text: `
+    await act(vi.runOnlyPendingTimersAsync)
+
+    const element = screen.getByText('bar')
+    expect(element)
+  })
+
+  test('render note when subthought is edited from non-attribute', async () => {
+    await dispatch([
+      importText({
+        text: `
       - a
         - note
           - foo`,
-    }),
-    setCursor(['a', 'note']),
-    (dispatch, getState) =>
-      dispatch(
-        editThought({
-          oldValue: 'note',
-          newValue: '=note',
-          path: getState().cursor as SimplePath,
-        }),
-      ),
-    // this will hide the meta attribute, so if the note value can be selected on the screen it must be rendered
-    setCursor(null),
-  ])
-
-  await act(vi.runOnlyPendingTimersAsync)
-
-  const element = screen.getByText('foo')
-  expect(element)
-})
+      }),
+      setCursor(['a', 'note']),
+      (dispatch, getState) =>
+        dispatch(
+          editThought({
+            oldValue: 'note',
+            newValue: '=note',
+            path: getState().cursor as SimplePath,
+          }),
+        ),
```

Comment:
We should avoid dispatching to the store directly, and instead use RTL to interact with the screen like the user. This is the main philosophy behind RTL. We try not to tightly couple tests to underlying infrastructure or implementation details.

Maybe we can use `focus` and `fireEvent` for editing?

FYI We currently dispatch `importText` directly to the store at the beginning of tests for simplicity, but [soon this will be refactored](https://github.com/cybersemics/em/issues/2980) to use `paste`, which is more user-centric and hides the global store access.